vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/BatchRegistrationDialog.java

```diff
@@ -99,60 +94,83 @@ private class RegisterBatchDialogHandler implements Serializable {
+          sampleSpreadsheetLayout.setBatchName(batchInformationLayout.batchNameField.getValue());
+        } else {
+          tabStepper.setSelectedTab(batchInformationTab);
+        }
+      });
+    }
+
+    private void setSubmissionListeners() {
+      setCancelSubmission();
+      setBatchRegistrationSubmission();
+      setSampleRegistrationSubmission();
+    }
+
+    private void setCancelSubmission() {
       batchInformationLayout.cancelButton.addClickListener(event -> cancelListeners.forEach(
           listener -> listener.onComponentEvent(
-              new UserCancelEvent<>(SampleRegistrationDialog.this))));
+              new UserCancelEvent<>(BatchRegistrationDialog.this))));
+      sampleSpreadsheetLayout.cancelButton.addClickListener(event -> cancelListeners.forEach(
+          listener -> listener.onComponentEvent(
+              new UserCancelEvent<>(BatchRegistrationDialog.this))));
     }
 
-    private void setSampleRegistrationSubmission() {
+    private void setBatchRegistrationSubmission() {
       sampleSpreadsheetLayout.registerButton.addClickListener(event -> {
         if (isInputValid()) {
-          listeners.forEach(listener -> listener.onComponentEvent(
-              new SampleRegistrationEvent(SampleRegistrationDialog.this, true)));
+          batchRegistrationListeners.forEach(listener -> listener.onComponentEvent(
+              new BatchRegistrationEvent(BatchRegistrationDialog.this, true)));
         }
       });
-      sampleSpreadsheetLayout.cancelButton.addClickListener(event -> cancelListeners.forEach(
-          listener -> listener.onComponentEvent(
-              new UserCancelEvent<>(SampleRegistrationDialog.this))));
+
+    }
+
+    private void setSampleRegistrationSubmission() {
+      sampleSpreadsheetLayout.registerButton.addClickListener(
+          event -> sampleRegistrationListeners.forEach(listener -> listener.onComponentEvent(
+              new SampleRegistrationEvent(BatchRegistrationDialog.this, true))));
     }
 
     protected boolean isInputValid() {
       return batchInformationLayout.isInputValid() && sampleSpreadsheetLayout.isInputValid();
     }
 
-    private void setTabSelectionListener() {
-      tabStepper.addSelectedChangeListener(event -> {
-        if (event.getSelectedTab() == sampleInformationTab
-            && batchInformationLayout.isInputValid()) {
-          sampleSpreadsheetLayout.generateSampleRegistrationSheet(
-              batchInformationLayout.dataTypeSelection.getValue());
-          sampleSpreadsheetLayout.setBatchName(batchInformationLayout.batchNameField.getValue());
-        } else {
-          tabStepper.setSelectedTab(batchInformationTab);
-        }
-      });
+
+    public void addBatchRegistrationEventListener(
+        ComponentEventListener<BatchRegistrationEvent> listener) {
+      this.batchRegistrationListeners.add(listener);
     }
 
     public void addSampleRegistrationEventListener(
```

Comment:
Why do we have batch registration events and sample registration events on the UI component level?